expo/MCTS.py

```diff
@@ -0,0 +1,457 @@
+import json
+import math
+import os
+import pickle
+import random
+import shutil
+
+import numpy as np
+import pandas as pd
+
+from expo.data.custom_task import get_mle_bench_requirements, get_mle_task_id
+from expo.data.dataset import generate_task_requirement, get_split_dataset_path
+from expo.evaluation.evaluation import evaluate_score
+from expo.insights.instruction_generator import InstructionGenerator
+from expo.research_assistant import ResearchAssistant, TimeoutException
+from expo.utils import get_exp_pool_path, load_execute_notebook, mcts_logger
+from metagpt.tools.tool_recommend import ToolRecommender
+from metagpt.utils.common import read_json_file
+
+
+def initialize_di_root_node(state, reflection: bool = True):
```

Comment:
Make appropriate type annotations, including all input parameters and all outgoing parameters.

Reply:
Other functions should be handled in a similar way.

Reply:
done